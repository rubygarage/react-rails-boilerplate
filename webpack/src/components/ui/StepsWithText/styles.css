@media (min-width: 768px) {
  .menu {
    display: flex;
    justify-content: center;
  }
}
