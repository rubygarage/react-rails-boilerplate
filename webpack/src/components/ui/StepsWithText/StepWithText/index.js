import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

import styles from './styles.css'

function StepWithText({ text, active, passed }) {
  return (
    <div className={classNames(styles.step, { [styles.active]: active, [styles.passed]: passed })}>
      <span className={styles['step-text']}>
        {text}
      </span>
    </div>
  )
}

StepWithText.propTypes = {
  active: PropTypes.bool,
  passed: PropTypes.bool,
  text: PropTypes.string
}

export default StepWithText
