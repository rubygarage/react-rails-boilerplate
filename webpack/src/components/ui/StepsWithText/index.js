import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import StepWithText from './StepWithText'

import styles from './styles.css'

function StepsWithText({ className, steps, step }) {
  const isActive = (index) => step === index + 1
  const isPassed = (index) => step > index + 1

  return (
    <div className={classNames(styles.menu, className)}>
      {steps.map((text, index) => (
        <StepWithText
          key={index}
          text={text}
          active={isActive(index)}
          passed={isPassed(index)}
        />
      ))}
    </div>
  )
}

StepsWithText.propTypes = {
  className: PropTypes.string,
  step: PropTypes.number,
  steps: PropTypes.array
}

export default StepsWithText
