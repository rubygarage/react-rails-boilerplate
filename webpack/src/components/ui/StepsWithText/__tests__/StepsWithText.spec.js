import React from 'react'
import { shallow } from 'enzyme'
import StepsWithText from 'components/ui/StepsWithText'

describe('<StepsWithText />', () => {
  const defaultProps = {
    steps: ['Invitation', 'Additional info', 'Membership'],
    step: 2
  }

  it('renders second step as active', () => {
    const component = shallow(
      <StepsWithText {...defaultProps} />
    )

    expect(component).toMatchSnapshot()
  })
})
