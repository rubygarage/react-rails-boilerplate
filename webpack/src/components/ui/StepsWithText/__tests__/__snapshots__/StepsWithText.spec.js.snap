// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StepsWithText /> renders second step as active 1`] = `
<div
  className="menu"
>
  <StepWithText
    active={false}
    passed={true}
    text="Invitation"
  />
  <StepWithText
    active={true}
    passed={false}
    text="Additional info"
  />
  <StepWithText
    active={false}
    passed={false}
    text="Membership"
  />
</div>
`;
